### PR TITLE
COMMERCE-10476 Fix the bug that didn't allow products with options to be added to quick cart

### PR DESCRIPTION
**Pull request by 22LucaGiannotta22**
Resent from [liferay-frontend#2961](https://github.com/liferay-frontend/liferay-portal/pull/2961)

https://issues.liferay.com/browse/COMMERCE-10476

### CODE REVIEW

**Comment by 22LucaGiannotta22** [comment]
ci:test:sf

**Comment by 22LucaGiannotta22** [comment]
ci:test:sf

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:test:sf

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase [hunk 1]
```diff
@@ -427,6 +427,81 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-10529. Buyer user can delete selected product chips from the search bar of the quick add-to-cart"
+	@priority = "3"
+	test CanDeleteSelectedSKUs {
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase:598]: CanDeleteQuickAddToCartChips

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase [hunk 2]
```diff
@@ -427,6 +427,81 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-10529. Buyer user can delete selected product chips from the search bar of the quick add-to-cart"
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase:596]: COMMERCE-10629

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase [hunk 3]
```diff
@@ -427,6 +427,81 @@ definition {
...
+		task ("And in the search bar no sku is left") {
+			AssertElementPresent(
+				key_searchBarValue = "Search Products",
+				locator1 = "CommerceAccelerators#QUICK_ADD_TO_CART_EMPTY_SEARCH_BAR");
+
+			for (var productSku : list "MIN55861,MIN55860,MIN55859") {
+				AssertElementNotPresent(
+					key_searchBarValue = "${productSku}",
+					locator1 = "CommerceAccelerators#QUICK_ADD_TO_CART_EMPTY_SEARCH_BAR");
+			}
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase:667]: for (var productSku : list "MIN55861,MIN55860,MIN55859") {
				AssertElementNotPresent(
					locator1 = "CommerceWidget#TAG_FACET_LABEL_LAYOUT",
					tagName = "${productSku}");
			}

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase [hunk 4]
```diff
@@ -523,6 +688,28 @@ definition {
...
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			Uncheck.uncheckToggleSwitch(
+				key_toggleSwitchLabel = "Purchasable",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Button.clickPublish();
+
+			CommerceUtilities.checkSuccessAlertAppearsAndDisappears();
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase:710]: An API can be used for this. Check _patchCommerceSku macro and PATCH
/v1.0/skus/{id}

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase [hunk 5]
```diff
@@ -551,49 +738,47 @@ definition {
...
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Purchasable",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Button.clickPublish();
+
+			CommerceUtilities.checkSuccessAlertAppearsAndDisappears();
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase:781]: An API can be used for this. Check _patchCommerceSku macro and PATCH
/v1.0/skus/{id}

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by 22LucaGiannotta22** [comment]
ci:test:sf

**Review by lorenzolapescara — APPROVED** [review]

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:commerce-functional

**Comment by kresimir-coko** [comment]
ci:forward

**Pull request closed without merging**